apps/openmw/mwmechanics/aisequence.cpp

```diff
@@ -274,17 +274,20 @@ void AiSequence::execute (const MWWorld::Ptr& actor, CharacterController& charac
         {
             if (package->execute (actor, characterController, mAiState, duration))
             {
-                // Put repeating noncombat AI packages on the end of the stack so they can be used again
+                // AiPackage::execute() can queue an another AI package, so we can not just use the mPackages.mfront() here.
+                std::list<MWMechanics::AiPackage*>::iterator toRemove =
+                        std::find(mPackages.begin(), mPackages.end(), package);
```

Comment:
Possibly at this point `mPackages` might be empty or `package` might not belong to the list. So find would return `end()`. `end()` can't be used for `erase`.